packages/crypto/src/Bip32/Bip32PrivateKey.ts

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable  no-bitwise */
+import * as Bip32KeyDerivation from './Bip32KeyDerivation';
+import { Bip32PublicKey } from './Bip32PublicKey';
+import { EXTENDED_ED25519_PRIVATE_KEY_LENGTH, Ed25519PrivateKey, NORMAL_ED25519_PRIVATE_KEY_LENGTH } from '../Ed25519e';
+import { InvalidArgumentError } from '../errors';
+import { crypto_scalarmult_ed25519_base_noclamp, ready } from 'libsodium-wrappers-sumo';
+import { pbkdf2 } from 'pbkdf2';
+
+const SCALAR_INDEX = 0;
+const SCALAR_SIZE = 32;
+const PBKDF2_ITERATIONS = 4096;
+const PBKDF2_KEY_SIZE = 96;
+const PBKDF2_DIGEST_ALGORITHM = 'sha512';
+const CHAIN_CODE_INDEX = 64;
+const CHAIN_CODE_SIZE = 32;
+
+/**
+ * clamp the scalar by:
+ *
+ *  1. clearing the 3 lower bits.
+ *  2. clearing the three highest bits.
+ *  3. setting the second-highest bit.
+ *
+ * @param scalar The clamped scalar.
+ */
+const clampScalar = (scalar: Buffer): Buffer => {
+  scalar[0] &= 0b1111_1000;
+  scalar[31] &= 0b0001_1111;
```

Comment:
why are we clearing the highest three bits? The paper says `the highest bit of the last byte is cleared`, not the "three highest bits".

Reply:
But that is due to another condition of the paper that says `If the third highest bit of the last byte of kL is not zero, discard ek.` So basically, if that third bit is not zero, we discard the key. Some libraries seem to _force_ that condition, but it is not what the paper says. I'll check what we use in the node. 

Would it be too much of a pain to _discard_ a key?

Reply:
Actually, I can't really understand why they say in the paper that the key should be discarded. Let me think about it.

Reply:
@AngelCastilloB can you capture this and any other issues in a new MD file, so we can close out this PR without missing any context. Once the audit is complete we can replace this file with a report.